### PR TITLE
NOISSUE - Improve twin definition editor ergonomy

### PR DESCRIPTION
**Pull request by darkodraskovic**
Improve twin definition editor ergonomy by enabling attribute selection from the list of already existing attributes and by using subtopic as attribute's name when attribute's name is empty.

### CODE REVIEW

File: src/app/pages/admin/twins/details/twins.details.component.ts [hunk 1]
```diff
@@ -64,9 +63,8 @@ export class TwinsDetailsComponent implements OnInit, OnDestroy {
     this.twinsService.getTwin(id).subscribe(
       resp => {
         this.twin = <Twin>resp;
-
+
```
Comment (manuio) [inline src/app/pages/admin/twins/details/twins.details.component.ts:66]: Remove these white spaces

File: src/app/pages/admin/twins/details/twins.details.component.ts [hunk 2]
```diff
@@ -192,15 +199,15 @@ export class TwinsDetailsComponent implements OnInit, OnDestroy {
       },
     };
     this.twinsService.editTwin(twin).subscribe(
-      resp => {
+      _resp => {
```
Comment (manuio) [inline src/app/pages/admin/twins/details/twins.details.component.ts:202]: There is a reason for this `_resp`?

File: src/app/pages/admin/twins/twins.component.ts [hunk 3]
```diff
@@ -112,32 +106,32 @@ export class TwinsComponent implements OnInit {
...
   }
 
-  onCreateConfirm(event): void {
+  onCreateConfirm(event: any): void {
     // close edditable row
     event.confirm.resolve();
 
     this.twinsService.addTwin(event.newData).subscribe(
-      resp => {
+      _resp => {
```
Comment (manuio) [inline src/app/pages/admin/twins/twins.component.ts:114]: same here

File: src/app/pages/admin/twins/twins.component.ts [hunk 4]
```diff
@@ -112,32 +106,32 @@ export class TwinsComponent implements OnInit {
...
-  onDeleteConfirm(event): void {
+  onDeleteConfirm(event: any): void {
     this.dialogService.open(ConfirmationComponent, { context: { type: 'twin' } }).onClose.subscribe(
       confirm => {
         if (confirm) {
           // close edditable row
           event.confirm.resolve();
           this.twinsService.deleteTwin(event.data.id).subscribe(
-            resp => {
+            _resp => {
```
Comment (manuio) [inline src/app/pages/admin/twins/twins.component.ts:134]: here

File: src/app/pages/admin/twins/twins.component.ts [hunk 5]
```diff
@@ -150,9 +144,9 @@ export class TwinsComponent implements OnInit {
     this.fsService.exportToCsv('twins.csv', this.twins);
   }
 
-  onFileSelected(files: FileList) {
+  onFileSelected(_files: FileList) {
```
Comment (manuio) [inline src/app/pages/admin/twins/twins.component.ts:147]: here

File: src/app/pages/admin/twins/twins.component.ts [hunk 6]
```diff
@@ -150,9 +144,9 @@ export class TwinsComponent implements OnInit {
...
   }
 
-  searchThing(input) {
+  searchThing(_input: any) {
```
Comment (manuio) [inline src/app/pages/admin/twins/twins.component.ts:150]: and here

File: src/app/pages/admin/twins/details/twins.details.component.ts (hunk 1 above)
Comment (drasko) [inline src/app/pages/admin/twins/details/twins.details.component.ts:66]: Please clean this whitespace

**Review by manuio — CHANGES_REQUESTED** [review]

**Review by drasko — CHANGES_REQUESTED** [review]

File: src/app/pages/admin/twins/details/twins.details.component.ts (hunk 2 above)
Comment (darkodraskovic) [reply src/app/pages/admin/twins/details/twins.details.component.ts:202]: Yes, it's a style standard to mark of the unused arguments.

Comment (darkodraskovic) [reply src/app/pages/admin/twins/details/twins.details.component.ts:202]: See here https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#flag-unused-declarations-with---nounusedparameters-and---nounusedlocals

File: src/app/pages/admin/twins/details/twins.details.component.ts (hunk 1 above)
Comment (darkodraskovic) [reply src/app/pages/admin/twins/details/twins.details.component.ts:66]: I need this line as logical separator.

File: src/app/pages/admin/twins/details/twins.details.component.ts (hunk 2 above)
Comment (manuio) [reply src/app/pages/admin/twins/details/twins.details.component.ts:202]: I'm not against but if you only fix it in a few places it's not consistent. Rather create an issue and fix it latter

File: src/app/pages/admin/twins/details/twins.details.component.ts (hunk 1 above)
Comment (manuio) [reply src/app/pages/admin/twins/details/twins.details.component.ts:66]: You need a new line not white spaces

**Review by manuio — APPROVED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**